flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -110,6 +117,8 @@ public NetworkEnvironment(
 
 		this.resultPartitionManager = new ResultPartitionManager();
 
+		this.inputGatesById = new ConcurrentHashMap<>();
```

Comment:
Which threads do access this structure concurrently?

Reply:
Canceler thread might close `SingleInputGate` while rpc thread is creating `SingleInputGate`.